packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -103,6 +119,52 @@ abstract class VirtualGatewayBase extends cdk.Resource implements IVirtualGatewa
       virtualGateway: this,
     });
   }
+
+  /**
+   * Grants the given entity all read permissions for this VirtualGateway.
+   */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:DescribeVirtualGateway',
+      'appmesh:ListVirtualGateway',
+      'appmesh:ListTagsForResource',
+    );
+  }
+
+  /**
+   * Grants the given entity all write permissions for this VirtualGateway.
+   */
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:CreateVirtualGateway',
+      'appmesh:UpdateVirtualGateway',
+      'appmesh:DeleteVirtualGateway',
+      'appmesh:TagResource',
+      'appmesh:UntagResource',
```

Comment:
Same questions here as for `GatewayRoute` - what would necessitate a resource having these permissions?